pyslim/slim_tree_sequence.py

```diff
@@ -490,6 +490,139 @@ def recapitate(self,
+
+        ``samples`` defaults to the number of samples present in the input tree sequence.
+
+        In general, all defaults excepting ``samples`` are whatever the defaults 
+        of ``msprime.simulate`` are; this includes recombination rate, so that if neither 
+        ``recombination_rate`` or a ``recombination_map`` are provided, there will be 
+        *no* recombination.
+
+        However, if ``recombination_rate`` *is* provided, then recapitation will
+        use a constant rate of recombination on a discretized map -- in other words,
+        recombinations in the coalescent portion of the simulation will only occur
+        at integer locations, just as in SLiM. If you do not want this to happen,
+        you need to construct a ``recombination_map`` explicitly.
+
+        Note that ``Ne`` defaults to ``1.0``; you probably
+        want to set it explicitly.
+
+        :param int slim_time: The desired period of time to simulate in msprime,
+            in units of generations
+        :param int Ne: The effective population size for :meth:`msprime.simulate`,
+            in units of diploid individuals
+        :param int samples: Sample count for :meth:`msprime.simulate`,
+            in units of haploid samples,
+        :param float mutation_rate: A (constant) mutation rate,
+            in units of mutations per nucleotide per unit of time.
+        :param float recombination_rate: A (constant) recombination rate,
+            in units of crossovers per nucleotide per unit of time.
+        :type recombination_map: :class`msprime.RecombinationMap`
+        :param recombination_map: The recombination map, or None,
+            if recombination_rate is specified.
+        :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+        '''
+        if "keep_first_generation" in kwargs:
+            raise ValueError("The keep_first_generation argument is deprecated:"
+                             "the FIRST_GEN flag is no longer used.")
+
+        discrete_msprime = hasattr(msprime, "RateMap")
+
+        if recombination_rate is not None:
+            if recombination_map is not None:
+                raise ValueError("Cannot specify length/recombination_rate along with a recombination map")
+            if discrete_msprime:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0])
+            else:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0],
+                                                             num_loci = int(self.sequence_length))
+        if discrete_msprime and ('discrete_genome' not in kwargs):
+            kwargs['discrete_genome'] = True
+
+        slim_indivs = self.individuals_alive_at(0)
```

Comment:
hm, if this becomes an msprime method we'll have to include another argument that says which nodes are the ones "alive at the end" (which could default to the samples)

Reply:
Yes this may be the only pyslim specific line. Wouldn't be too tough to implement though!